src/analysis/mean_inequalities.lean

```diff
@@ -964,6 +964,97 @@ begin
   exact lintegral_Lp_add_le_aux hpq hf hf_top hg hg_top h0 htop,
 end
 
+lemma add_rpow_le_one_of_add_le_one {p : ℝ} (a b : ennreal) (hab : a + b ≤ 1) (hp1 : 1 ≤ p) :
+  a ^ p + b ^ p ≤ 1 :=
+begin
+  have h_le_one : ∀ x : ennreal, x ≤ 1 → x ^ p ≤ x,
+  { intros x hx,
+    nth_rewrite 1 ←ennreal.rpow_one x,
+    exact ennreal.rpow_le_rpow_of_exponent_ge hx hp1, },
+  have ha : a ≤ 1,
+  { calc a ≤ a + b : self_le_add_right a b
+      ...  ≤ 1     : hab, },
```

Comment:
When a computation like that is trivial, you can shorten it to
```suggestion
  have ha : a ≤ 1 := (self_le_add_right a b).trans hab,
```
Same thing for next line.